lib/client/signal_fx_client.js

```diff
@@ -134,6 +207,24 @@ SignalFxClient.prototype.startAsyncSend = function () {
   });
 };
 
+SignalFxClient.prototype.startAsyncEventSend = function () {
+  if (this.loadingAWSUniqueId) {
+    return;
```

Comment:
Wouldn't this cause a .send() immediately following instantiation of the SignalFxClient to fail silently? It looks like this is just doing a noop if .send is called which the AWS unique id is being loaded. It should wait for the unique id to be loaded and then complete the .send request.